### ISSUE TITLE
Excessive error message context

### ISSUE THREAD

**Issue by yunal** [chef/chef#12001]
## Description

Chef-client 17.4.38 provides 519MB of context information in the terminal on some failures in our environment.

Since I use the open source cinc I filed a bug there first. They wanted me to file it upstream.  https://gitlab.com/cinc-project/distribution/client/-/issues/37. I repeat the information here:

## Chef Version

$ chef-client --version
Redirecting to cinc-client
Cinc Client: 17.4.38

## Platform Version

Debian 11, fresh, updated install on a VM.

## Replication Case

I was fixing a php related recipe and made a mistake, I sent a reload notification to a execute resource:

```
Recipe: php::default
  * execute[phpenmod timezone] action reload

================================================================================
    Error executing action `reload` on resource 'execute[phpenmod 
timezone]'
================================================================================
```

The resulting error message started like this:

```
   NoMethodError
    -------------
    undefined method `action_reload' for 
#<Chef::Provider::Execute:0x0000559fb8683af0 
@new_resource=<execute[phpenmod timezone] @name: "phpenmod timezone" 
@before: nil @params: {} @provider: nil @allowed_actions: [:nothing, 
:run] @action: [:nothing] @updated: false @updated_by_last_action: 
false @source_line: 
"/var/cache/chef/cookbooks/php/recipes/default.rb:45:in `from_file'" 
@guard_interpreter: nil @default_guard_interpreter: :execute 
@elapsed_time: 0 @command: "phpenmod timezone" @is_guard_interpreter: 
false @declared_type: :execute @cookbook_name: "php" @recipe_name: 
"default" @domain: nil @user: nil>, @action=:reload, 
@current_resource=nil, @after_resource=nil, 
@run_context=#<Chef::RunContext:0x0000559fb52263c0 
@events=#<Chef::EventDispatch::Dispatcher:0x
```

which seems reasonable.  But then it went on a bit.  I had to capture to a file:

```
# ls -l --si foo.log
-rw-r--r-- 1 root root 519M aug.  26 14:39 foo.log
```

## Client Output

See above. I cannot provide the full log because it's unreasonable to vet for security issues.

## Stacktrace

See above.

**Comment by yunal** [comment]
```
log_level          :warn
```
btw

**Comment by lamont-granquist** [comment]
Yeah this is massive and across the board.  I've fixed it for a few new objects that I'm writing in #11904 but the approach there needs to be extracted into some kind of mixin/DSL with probably a slightly nicer API than the rage-code one I threw together in that PR and then applied more generally.

Fundamentally its just because the run_context and/or node tends to be an instance variable off of most objects and the inspect() method will iterate recursively through all instance variables and all their instances variables which inevitably prints the whole node object and resource collection.

Anyone who wants to try to fix this one should probably talk to me first about exactly what I want to see in a design/PR.
